Content.Server/_NF/Speech/EntitySystems/GoblinAccentSystem.cs

```diff
@@ -1,31 +1,59 @@
 using Content.Server.Speech;
 using Content.Server.Speech.EntitySystems;
+using System.Text.RegularExpressions;
 
 namespace Content.Server._NF.Speech.EntitySystems;
 
+// The whole code is a copy of SouthernAccentSystem by UBlueberry (https://github.com/UBlueberry)
 public sealed class GoblinAccentSystem : EntitySystem
 {
+    private static readonly Regex RegexIng = new(@"ing\b");
+    private static readonly Regex RegexAnd = new(@"\band\b");
+    private static readonly Regex RegexEr = new(@"er\b");
+    private static readonly Regex RegexTt = new(@"tt");
+    private static readonly Regex RegexOf = new(@"\bof\b");
+    private static readonly Regex RegexTo = new(@"\bto\b");
+    private static readonly Regex RegexThe = new(@"\bthe\b");
+    private static readonly Regex RegexH = new(@"\bh");
+    private static readonly Regex RegexSelf = new(@"self\b");
+    private static readonly Regex RegexLe = new(@"le\b");
+    private static readonly Regex RegexAll = new(@"all");
+    private static readonly Regex RegexAl = new(@"al");
+    private static readonly Regex RegexUl = new(@"ul");
+    private static readonly Regex RegexEll = new(@"ell\b");
+    private static readonly Regex RegexWl = new(@"wl");
+
     [Dependency] private readonly ReplacementAccentSystem _replacement = default!;
 
     public override void Initialize()
     {
         base.Initialize();
-
-        SubscribeLocalEvent<GoblinAccentComponent, AccentGetEvent>(OnAccentGet);
+        SubscribeLocalEvent<GoblinAccentComponent, AccentGetEvent>(OnAccent);
     }
 
-    // converts left word when typed into the right word. For example typing you becomes ye.
-    public string Accentuate(string message, GoblinAccentComponent component)
+    private void OnAccent(EntityUid uid, GoblinAccentComponent component, AccentGetEvent args)
     {
-        var msg = message;
+        var message = args.Message;
 
-        msg = _replacement.ApplyReplacements(msg, "goblin");
-        return msg;
-    }
+        message = _replacement.ApplyReplacements(message, "goblin");
 
-    private void OnAccentGet(EntityUid uid, GoblinAccentComponent component, AccentGetEvent args)
-    {
-        args.Message = Accentuate(args.Message, component);
+        message = RegexIng.Replace(message, "in'");
+        message = RegexAnd.Replace(message, "an'");
+        message = RegexEr.Replace(message, "ah");
+        message = RegexTt.Replace(message, "'");
+        message = RegexH.Replace(message, "'");
+        message = RegexSelf.Replace(message, "seuf");
+        message = RegexOf.Replace(message, "o'");
+        message = RegexTo.Replace(message, "ta");
+        message = RegexThe.Replace(message, "da");
+        message = RegexLe.Replace(message, "ow");
+        message = RegexAll.Replace(message, "aw");
+        message = RegexAl.Replace(message, "aw");
+        message = RegexUl.Replace(message, "w");
+        message = RegexEll.Replace(message, "ew");
+        message = RegexWl.Replace(message, "w");
```

Comment:
These phonetic replacements are almost all problematic (and I have a huge problem with how the Southern accent is implemented too, don't use that in your defence!):

- -ing is only really reduced to -in as part of the verb ending. Counterexamples: thing ⇏ thin, bring ⇏ brin, ring ⇏ rin, king ⇏ kin
- -er ⇒ -ah is troublesome for a few small function words. Counterexamples: her ⇏ hah, per ⇏ pah, er ⇏ ah (e.g. "er okay" turns into "ah okay" – a drastic departure in meaning).
- -er ⇒ -ah is also troublesome when the letter preceding the ⟨e⟩ is a soft ⟨c⟩ or ⟨g⟩: nicer ⇒ nicah (looks like /s/ became /k/), officer ⇒ officah (same), charger ⇒ chargah (looks like /dʒ/ turned into /g/)
- [T-glottalisation](https://en.wikipedia.org/wiki/T-glottalization) is orders of magnitude more complex than just replacing tt with an apostrophe. It's heavily contextual. For instance, it may or may not happen when /r/ follows: mattress ⇏ ma'ress (sometimes).
- "seuf" looks like a rogue French word. If you really want to play into the L-into-W meme, consider being consistent with yourself and choosing "sewf".
- All the L-to-W things: this process is _enormously complicated_. It generally does not happen intervocallically, i.e., between vowels, and doesn't always become a /w/ sound. These replacements turn things into a right mess sometimes. Counterexamples:
    * allow ⇏ awow (wrong vowel, also intervocalic)
    * bottle of water ⇏ bo'ow o' watah (the vowel in 'of' causes the L to remain)
    * pal ⇏ paw (looks like /æ/ turns into /ɔ/)
    * gal ⇏ gaw (same)
    * multiple ⇏ mwtipow (yikes)
    * balance ⇏ bawance
- also, ⟨ow⟩ has a common reading of /aʊ/ (h<ins>ow</ins> n<ins>ow</ins> br<ins>ow</ins>n c<ins>ow</ins>) and isn't a super suitable substitute for /ɫ/.
- to ⇒ ta is _going to cause problems_ given that (1) half the people writing English don't know the difference between 'to' and 'too', (2) 'ta' is already a word, chiefly used in Britain, that means 'thanks'.

Ta muchly for reading. My general stance on phonetic replacements is: don't do 'em unless you're trained in linguistics, in which case you'll realise the futility of even attempting to do them on a grapheme basis.

Reply:
ngl this attempt at simulating cockney pronunciation was doomed from the start.

I have no issues with removing 'l'-related bits, but I don't think I'll remove 'ing' or 'er' because it is unreasonable to try and compile an endless list of words where it'll be appropriate to replace ing ⇒ in' or er ⇒ schwa. I think (I don't know if it is the case) that there are far more verbs ending with 'ing' than nouns that happen to have 'ing' at the end. and as such, I believe blanket replacement of all '-ing' and '-er' with -in' and -ah is just "good enough".

I could probably add a couple of more commonly used nouns and whatnot that end with 'ing' to the table, because I think that words in chat messages are replaced with words from the table first, and what was left after that is replaced with bits in C#.

> [T-glottalisation](https://en.wikipedia.org/wiki/T-glottalization) is orders of magnitude more complex than just replacing tt with an apostrophe. It's heavily contextual. For instance, it may or may not happen when /r/ follows: mattress ⇏ ma'ress (sometimes).

Even though the subject matter is more complex and nuanced, the 'tt' stuff is just about the only thing that could be easily handled with C#.

> (2) 'ta' is already a word, chiefly used in Britain, that means 'thanks'.

Damn. Polite goblins. Can't have that.

> linguistics ... grapheme 

Long words. Me no know.

Thank you for taking the time to share bits of linguistic knowledge.